src/mc_cell_group.hpp

```diff
@@ -210,9 +227,52 @@ class mc_cell_group: public cell_group {
         PL();
     }
 
-    void enqueue_events(const std::vector<postsynaptic_spike_event>& events) override {
-        for (auto& e: events) {
-            events_.push(e);
+    void enqueue_events(
+            util::subrange_view_type<std::vector<std::vector<postsynaptic_spike_event>>> events,
+            time_type tfinal,
+            size_t epoch) override
+    {
+        using pse = postsynaptic_spike_event;
+
+        // Make convenience variables for event lanes:
+        //  lf: event lanes for the next epoch
+        //  lc: event lanes for the current epoch
+        auto& lf = event_lanes(epoch+1);
+        const auto& lc = event_lanes(epoch);
+
+        // For a cell, merge the incoming events with events in lc that are
+        // not to be delivered in thie epoch, and store the result in lf.
+        auto merge_future_events = [&](unsigned l) {
+            PE("sort");
+            // STEP 1: sort events in place in events[l]
+            util::sort_by(events[l], [](const pse& e){return event_time(e);});
+            PL();
+
+            PE("merge");
+            // STEP 2: clear lf to store merged list
+            lf[l].clear();
+
+            // STEP 3: merge new events and future events from lc into lf
+            auto pos = std::lower_bound(lc[l].begin(), lc[l].end(), tfinal, event_time_less());
```

Comment:
Quite often we have to write functors of this type to work with events and spikes. I find it distracting to have to keep creating them when needed.

I prefer to stick with a `std::vector` than go with a specialised data structure. Or are do you mean a specialised data structure for both queues and their associated methods?

Reply:
Regarding the comparator, I have no objection to the generic code, just thinking that it could live in the same file as the code that uses it, at least until it becomes useful elsewhere.

I was thinking of the data structure that wraps the per-epoch per-cell event collections; if we move the machinery up to the model level as we were discussing earlier, than that would side-step the issue.